test/local-hub/external_storage_backend_test.go

```diff
@@ -0,0 +1,580 @@
+		OwnerID: "demo/testing",
+	})
+	require.NoError(t, err)
+	familyDetails, err = cli.ListTypeInstances(ctx, &gqllocalapi.TypeInstanceFilter{
+		CreatedBy: ptr.String("nature"),
+	}, local.WithFields(local.TypeInstanceAllFields))
+	require.NoError(t, err)
+
+	assertTypeInstancesDetail(t, familyDetails, family, getDataDirectlyFromStorage(t, srvAddr, familyDetails), []*expectedTypeInstanceData{
+		{
+			alias: ptr.String("second-child"),
+			backendContext: map[string]interface{}{
+				"updated": "context 0",
+			},
+			dataInGQL: map[string]interface{}{
+				"updated-value": "context 0",
+			},
+			dataInExternalBackend: map[string]interface{}{
+				"updated-value": "context 0",
+			},
+			locked:                  "",
+			lockedInExternalBackend: "",
+		},
+		{
+			alias: ptr.String("child"),
+			backendContext: map[string]interface{}{
+				"updated": "context 1",
+			},
+			dataInGQL: map[string]interface{}{
+				"updated-value": "context 1",
+			},
+			dataInExternalBackend:   nil,
+			locked:                  "",
+			lockedInExternalBackend: "",
+		},
+		{
+			alias:          ptr.String("original"),
+			backendContext: nil,
+			dataInGQL: map[string]interface{}{
+				"name": "Anakin Skywalke",
+			},
+			dataInExternalBackend: map[string]interface{}{
+				"name": "Anakin Skywalke",
+			},
+			locked:                  "",
+			lockedInExternalBackend: "",
+		},
+		{
+			alias: ptr.String("parent"),
+			backendContext: map[string]interface{}{
+				"provider": "dotenv",
+			},
+			dataInGQL: map[string]interface{}{
+				"name": "Darth Vader",
+			},
+			dataInExternalBackend: map[string]interface{}{
+				"name": "Darth Vader",
+			},
+			locked:                  "",
+			lockedInExternalBackend: "",
+		},
+	})
+}
+
+func registerExternalDotenvStorage(ctx context.Context, t *testing.T, cli *local.Client, srvAddr string) (gqllocalapi.CreateTypeInstanceOutput, func()) {
+	t.Helper()
+
+	ti, err := cli.CreateTypeInstances(ctx, fixExternalDotenvStorage(srvAddr))
+	require.NoError(t, err)
+	require.Len(t, ti, 1)
+	dotenvHubStorage := ti[0]
+
+	return dotenvHubStorage, func() {
+		_ = cli.DeleteTypeInstance(ctx, dotenvHubStorage.ID)
+	}
+}
+
+func fixExternalDotenvStorage(addr string) *gqllocalapi.CreateTypeInstancesInput {
+	return &gqllocalapi.CreateTypeInstancesInput{
+		TypeInstances: []*gqllocalapi.CreateTypeInstanceInput{
+			{
+				CreatedBy: ptr.String("manually"),
+				TypeRef: &gqllocalapi.TypeInstanceTypeReferenceInput{
+					Path:     "cap.type.example.filesystem.storage",
+					Revision: "0.1.0",
+				},
+				Value: storageSpec{
+					URL:         ptr.String(addr),
+					AcceptValue: ptr.Bool(true),
+					ContextSchema: ptr.String(heredoc.Doc(`
+				      {
+				      	"$id": "#/properties/contextSchema",
+				      	"type": "object",
+				      	"properties": {
+				      		"provider": {
+				      			"$id": "#/properties/contextSchema/properties/name",
+				      			"type": "string",
+								  "enum": [
+									"aws_secretsmanager",
```

Comment:
🤔  Why do the tests pass for creating TypeInstances without context, if our schema says it is required? We do validate the context, right? Is your PR rebased?

Reply:
Actually, if we create the TypeInstance without context then the default context is used (for use is the `dotenv`).  [Here](https://github.com/capactio/capact/blob/2939c70800f9f03ea95011646d0a39df84595de0/hub-js/src/local/storage/service.ts#L468) is a place where it is checked.